src/Microsoft.ML.Core/Utilities/OrderedWaiter.cs

```diff
@@ -90,7 +90,7 @@ public void Wait(long position, CancellationToken token = default(CancellationTo
                 ev = new WaitStats(position);
                 _waiters.Add(ev);
             }
-            ev.Event.Wait(token);
+            ev.Event.Wait(5*60*1000, token);
```

Comment:
📝 This change should be reverted. This is production code, not test code. If you want to add a delay, update the tests to pass a `CancellationToken` that will cancel after a period of time.